pkg/command/root.go

```diff
@@ -86,20 +88,27 @@ func NewRootCmd() (*cobra.Command, error) {
 				// is `Context-Scoped` then the new context-scoped plugin gets higher precedence.
 				// We therefore replace the existing command with the new command by removing the old and
 				// adding the new one.
-				maskedPlugins = append(maskedPlugins, matchedCmd.Name())
+				maskedPluginsWithPluginOverlap = append(maskedPluginsWithPluginOverlap, matchedCmd.Name())
 				rootCmd.RemoveCommand(matchedCmd)
 				rootCmd.AddCommand(cmd)
 			} else if plugins[i].Name != "login" {
 				// As the `login` plugin is now part of the core Tanzu CLI command and not a plugin
 				// anymore, skip the `login` plugin from adding it to the maskedPlugins array to avoid
 				// the warning message from getting shown to the user on each command invocation.
-				maskedPlugins = append(maskedPlugins, plugins[i].Name)
+				if isPluginCommand(matchedCmd) {
+					maskedPluginsWithPluginOverlap = append(maskedPluginsWithPluginOverlap, plugins[i].Name)
+				} else {
+					maskedPluginsWithCoreCmdOverlap = append(maskedPluginsWithCoreCmdOverlap, plugins[i].Name)
+				}
 			}
 		}
 	}
 
-	if len(maskedPlugins) > 0 {
-		fmt.Fprintf(os.Stderr, "Warning, Masking commands for plugins %q because a core command or other plugin with that name already exists. \n", strings.Join(maskedPlugins, ", "))
+	if len(maskedPluginsWithPluginOverlap) > 0 {
+		catalog.DeleteIncorrectPluginEntriesFromCatalog()
+	}
+	if len(maskedPluginsWithCoreCmdOverlap) > 0 {
+		fmt.Fprintf(os.Stderr, "Warning, Masking commands for plugins %q because a core command with that name already exists. \n", strings.Join(maskedPluginsWithCoreCmdOverlap, ", "))
```

Comment:
Since we are changing this, might as well fix the string:
```suggestion
		fmt.Fprintf(os.Stderr, "Warning, masking commands for plugins %q because a core command with that name already exists. \n", strings.Join(maskedPluginsWithCoreCmdOverlap, ", "))
```